x-pack/plugins/observability_solution/investigate_app/public/pages/details/components/assistant_hypothesis/assistant_hypothesis.tsx

```diff
@@ -37,9 +44,34 @@ export function AssistantHypothesis({ investigationId }: { investigationId: stri
 
   const serviceName = alert?.['service.name'] as string | undefined;
 
-  const [events, setEvents] = useState<RootCauseAnalysisForServiceEvent[]>([]);
+  const [events, setEvents] = useState<RootCauseAnalysisForServiceEvent[]>(
+    investigation?.automatedRcaAnalysis?.events ?? []
+  );
+
   const [loading, setLoading] = useState(false);
 
+  useEffect(() => {
+    const updateInvestigationSO = async () => {
+      if (investigation && events.length > 0) {
+        if (
+          events.find(
+            (event) =>
+              'response' in event && 'report' in event.response && 'timeline' in event.response
+          )
+        ) {
+          await updateInvestigation({
```

Comment:
Sorry, I missed this, but we need to do it on the server, not on the client. Otherwise the user needs to keep the browser tab open.

Reply:
Yes that is the whole point, if the user runs it, they shouldn't be required to keep the browser tab open.

Reply:
The reason for storing LLM result was that when you visit the page again, you don't have to run the analysis and wait for few minutes. But when you run the analysis for the first time, you would want to see the result, isn't it? So imo it's ok to keep the browser tab open whenever new analysis is being run. Another aspect is that if user closes the tab, that would mean they don't want to continue with the analysis, right? We should not carry on with the requests to LLM in that case.

Reply:
No, as I said before, the point is that the user needs to be able to navigate away from the page and the results will be ready when they get back. Why ask them to keep a browser tab open for minutes and make sure they stay on the same actual page?